### PR TITLE
7994-enhance-mlpblock

### LINKED ISSUE #7994: `MLPBlock` cannot support tuple input of GEGLU

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Fixes #7994  .

### Description
The current implementation does not support tuple input of "GEGLU" since it only change the out features of the first linear layer when the input is a string of "GEGLU".

This PR enhances it, and also enable "vista3d" mode to support #7987 
Tests are added to cover the changes.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_mlp.py [hunk 1]
```diff
@@ -31,6 +33,14 @@
             ]
             TEST_CASE_MLP.append(test_case)
 
+# test different activation layers
+TEST_CASE_ACT = []
+for act in ["GELU", "GEGLU", ("GELU", {"approximate": "tanh"}), ("GEGLU", {})]:
```
Comment (yiheng-wang-nv) [inline tests/test_mlp.py:38]: without the changes, `("GEGLU", {})` cannot pass tests because of mismatch output features of linear1 cc @KumoLiu

**Comment by KumoLiu** [comment]
@yiheng-wang-nv, thanks for the pr, overall looks good to me.
please help fix the mypy issue: https://github.com/Project-MONAI/MONAI/actions/runs/10261750234/job/28390075552?pr=7995
Thanks.

**Review by KumoLiu — APPROVED** [review]

**Comment by KumoLiu** [comment]
```
[2024-08-06T12:00:46.957Z] ======================================================================
[2024-08-06T12:00:46.957Z] ERROR: test_act_2 (tests.test_mlp.TestMLPBlock)
[2024-08-06T12:00:46.957Z] ----------------------------------------------------------------------
[2024-08-06T12:00:46.957Z] Traceback (most recent call last):
[2024-08-06T12:00:46.957Z]   File "/usr/local/lib/python3.9/dist-packages/parameterized/parameterized.py", line 620, in standalone_func
[2024-08-06T12:00:46.957Z]     return func(*(a + p.args), **p.kwargs, **kw)
[2024-08-06T12:00:46.957Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_mlp.py", line 60, in test_act
[2024-08-06T12:00:46.957Z]     net = MLPBlock(**input_param)
[2024-08-06T12:00:46.957Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/networks/blocks/mlp.py", line 57, in __init__
[2024-08-06T12:00:46.957Z]     self.fn = get_act_layer(act)
[2024-08-06T12:00:46.957Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/networks/layers/utils.py", line 75, in get_act_layer
[2024-08-06T12:00:46.957Z]     return act_type(**act_args)
[2024-08-06T12:00:46.957Z] TypeError: __init__() got an unexpected keyword argument 'approximate'
[2024-08-06T12:00:46.957Z] 
[2024-08-06T12:00:46.957Z] ----------------------------------------------------------------------
[2024-08-06T12:00:46.957Z] Ran 14447 tests in 904.177s
```
@yiheng-wang-nv please help take a look at the ci error.

**Comment by KumoLiu** [comment]
https://github.com/Project-MONAI/MONAI/actions/runs/10277832292/job/28440486412?pr=7995#step:7:370

**Pull request merged**